### PR TITLE
Support qualifiers for ambiguous plugin commands

### PR DESCRIPTION
**Pull request by neonichu**
This adds a `--package` option to the plugin command which will limit the set of available plugins to those inside a package with the given identity.

The identity is not validated in any way, it will just be applied as a filter on all relevant packages. The option does apply to `--list` as well which seems a bit odd but a consequence of making listing an option rather than a command.

rdar://99887952

Previously:

```
❯ swift-package plugin B --help            
error: 2 plugins found for ‘B’
Usage: swift package <options> <subcommand>
  See 'package -help' for more information.
```

Now:

```
❯ swift-package plugin --package B B --help
Hello, World!
```

### CODE REVIEW

**Review by MaxDesiatov — APPROVED** [review]
Obligatory "can this be tested?" question 🙈

**Comment by neonichu** [comment]
> Obligatory "can this be tested?" question 🙈

It should probably :)

**Comment by neonichu** [comment]
@swift-ci please test

**Comment by neonichu** [comment]
@swift-ci please test

**Comment by neonichu** [comment]
@swift-ci please test windows

File: Sources/Commands/PackageTools/PluginCommand.swift [hunk 1]
```diff
@@ -386,3 +392,13 @@ extension SandboxNetworkPermission {
         }
     }
 }
+
+extension ResolvedPackage {
+    fileprivate func matching(identity: String?) -> Bool {
+        if let identity {
+            return self.identity.description == identity.lowercased()
```
Comment (tomerd) [inline Sources/Commands/PackageTools/PluginCommand.swift:399]: consider `self.identity == .plain(identity)`

File: Sources/Commands/PackageTools/SwiftPackageTool.swift [hunk 2]
```diff
@@ -126,7 +126,7 @@ extension PluginCommand.PluginOptions {
     func merged(with other: Self) -> Self {
         // validate against developer mistake
         assert(
-            Mirror(reflecting: self).children.count == 3,
+            Mirror(reflecting: self).children.count == 4,
```
Comment (tomerd) [inline Sources/Commands/PackageTools/SwiftPackageTool.swift:129]: \o/

**Review by tomerd — APPROVED** [review]

**Comment by neonichu** [comment]
@swift-ci please test

**Comment by neonichu** [comment]
@swift-ci please test windows

**Pull request merged**